From e269aa0adb20c60a745710e366becaa202ce7d0d Mon Sep 17 00:00:00 2001
From: Tanmay Jagdale <tanmayj@broadcom.com>
Date: Thu, 27 Nov 2014 09:27:03 +0530
Subject: xlp-config: Set device frequency to 0

Set the device frequencies to 0 so that their default value is setup
by the fdtconf kernel module. If a value other than zero is set, the
module does not change the value.
Also removed the frequency value in generic-nae.dtsi since its not used.
[Based on SDK 3.2]
Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
Signed-off-by: Nam Ninh <nam.ninh@windriver.com>

diff --git a/arch/mips/boot/dts/generic-nae.dtsi b/arch/mips/boot/dts/generic-nae.dtsi
index ad2cc8a..839e2e5 100644
--- a/arch/mips/boot/dts/generic-nae.dtsi
+++ b/arch/mips/boot/dts/generic-nae.dtsi
@@ -3,7 +3,6 @@ nae@node-0 {
 	compatible = "NETL,XLP8XX_A0";
 	#address-cells = <1>;
 	#size-cells = <1>;
-	frequency = <500>;
 
 	/*
 	 * Create a skeleton structure to load the ucore code at runtime.
@@ -532,7 +531,6 @@ nae@node-1 {
 	compatible = "NETL,XLP8XX_A0";
 	#address-cells = <1>;
 	#size-cells = <1>;
-	frequency = <500>;
 
 	/*
 	 * Create a skeleton structure to load the ucore code at runtime.
diff --git a/arch/mips/boot/dts/xlp-config.dts b/arch/mips/boot/dts/xlp-config.dts
index 03f1c6f..7e5562a 100644
--- a/arch/mips/boot/dts/xlp-config.dts
+++ b/arch/mips/boot/dts/xlp-config.dts
@@ -213,10 +213,15 @@
 	};
 
 	frequency-config {
-		nae = <500>;
-		sae = <500>;
-		rsa = <500>;
-		dtre = <500>;
-		cde = <250>;
+		/*
+		 * All device frequency parameters are set 0 and get initialized to the default values when the
+		 * 'fdtconf' kernel module is loaded at runtime.
+		 * If a value other than 0 is specified, the value is not modified by the module.
+		 */
+		nae = <0>;	/* Default - 500 Mhz (400 Mhz for XLP2XX) */
+		sae = <0>;	/* Default - 500 Mhz */
+		rsa = <0>;	/* Default - 500 Mhz */
+		dtre = <0>;	/* Default - 500 Mhz */
+		cde = <0>;	/* Default - 250 Mhz */
 	};
 };
diff --git a/arch/mips/boot/dts/xlp9xx-config.dts b/arch/mips/boot/dts/xlp9xx-config.dts
index 1994157..f02c898 100644
--- a/arch/mips/boot/dts/xlp9xx-config.dts
+++ b/arch/mips/boot/dts/xlp9xx-config.dts
@@ -279,10 +279,15 @@
 	};
 
 	frequency-config {
-		nae = <500>;
-		sae = <500>;
-		rsa = <500>;
-		dtre = <250>;
-		cde = <250>;
+		/*
+		 * All device frequency parameters are set 0 and get initialized to the default values when the
+		 * 'fdtconf' kernel module is loaded at runtime.
+		 * If a value other than 0 is specified, the value is not modified by the module.
+		 */
+		nae = <0>;	/* Default - 500 Mhz (400 Mhz for XLP2XX) */
+		sae = <0>;	/* Default - 500 Mhz */
+		rsa = <0>;	/* Default - 500 Mhz */
+		dtre = <0>;	/* Default - 500 Mhz */
+		cde = <0>;	/* Default - 250 Mhz */
 	};
 };
-- 
1.7.1

